platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/HalfDiskHashMap.java

```diff
@@ -127,18 +128,28 @@ public class HalfDiskHashMap<K extends VirtualKey>
      */
     private Thread writingThread;
 
-    /** MerkleDb settings */
-    private static final MerkleDbConfig config = ConfigurationHolder.getConfigData(MerkleDbConfig.class);
-
     /** Executor for parallel bucket reads/updates in {@link #endWriting()} */
-    private static final ExecutorService flushExecutor = Executors.newFixedThreadPool(
-            config.getNumHalfDiskHashMapFlushThreads(),
-            new ThreadConfiguration(getStaticThreadManager())
-                    .setComponent(MERKLEDB_COMPONENT)
-                    .setThreadName("HalfDiskHashMap Flushing")
-                    .setExceptionHandler((t, ex) ->
-                            logger.error(EXCEPTION.getMarker(), "Uncaught exception during HDHM flushing", ex))
-                    .buildFactory());
+    private static final AtomicReference<ExecutorService> flushExecutor = new AtomicReference<>();
+
+    private static ExecutorService getFlushExecutor() {
+        ExecutorService exec = flushExecutor.get();
+        if (exec == null) {
```

Comment:
Minor: same comment as above. Since we use this pattern in a few places, perhaps we could create an utility holder class that accepts a Supplier<> and implements this behavior in a good way, and then just reuse it.